src/main/scopes/js/metrics/function-metric.ts

```diff
@@ -71,24 +74,48 @@ export class FunctionMetric extends ScopeMetric {
       this.instrumentedPackage.version
     )
 
-    // convert arguments into a fake object to satisfy the substitute api
-    const argsObj: Record<string, unknown> = this.jsFunction.arguments.reduce(
-      (cur, val, index) => ({ ...cur, [index.toString()]: val }),
-      {}
-    )
+    let functionName = this.jsFunction.name
+    const functionAccessPath = this.jsFunction.accessPath
 
-    const anonymizedArguments = substitute(
-      argsObj,
-      Object.keys(argsObj), // all keys are allowed
-      this.allowedArgumentStringValues
-    )
+    // Handle renamed functions
+    if (this.matchingImport.rename !== undefined) {
+      functionName = functionName.replace(this.matchingImport.rename, this.matchingImport.name)
+      // replace the import name in access path
+      functionAccessPath[0] = this.matchingImport.name
+    }
+
+    const subs = new Substitution()
+    // redact complex values
+    functionAccessPath.forEach((segment) => {
+      if (segment instanceof ComplexValue) {
+        functionName = safeStringify(functionName).replace(
+          safeStringify(segment.complexValue),
+          subs.put(segment)
+        )
+      }
+    })
+
+    // redact "all" import
+    if (this.matchingImport.isAll) {
+      functionAccessPath[0] = subs.put(this.matchingImport.name)
+      functionName = functionName.replace(
+        this.matchingImport.name,
+        subs.put(this.matchingImport.name)
+      )
+    }
 
     let metricData: Attributes = {
-      [JsScopeAttributes.FUNCTION_NAME]: this.jsFunction.name,
-      [JsScopeAttributes.FUNCTION_ACCESS_PATH]: this.jsFunction.accessPath,
-      [JsScopeAttributes.FUNCTION_ARGUMENT_VALUES]: Object.values(anonymizedArguments).map((arg) =>
-        String(arg)
-      ),
+      [JsScopeAttributes.FUNCTION_NAME]: functionName,
+      // TODO: uncomment
+      // [JsScopeAttributes.FUNCTION_MODULE_SPECIFIER]: this.matchingImport.path,
+      [JsScopeAttributes.FUNCTION_ACCESS_PATH]: substituteArray(
+        functionAccessPath,
+        functionAccessPath.filter((p) => typeof p === 'string')
```

Comment:
Am I correct in understanding that what this is doing is to allow any non-complex value from the access path under the assumption that for this type of metric, the access path would have to be entirely from sub-parts of an instrumented function?

Reply:
Correct, the accessPath of a JsFunction can only be a string or a ComplexValue, as per defined by the [interface](https://github.com/ibm-telemetry/telemetry-js/pull/216/files#diff-57923b99fcb2d80f0d95a95d5f5b0adc4a1c74c69ab551abf8dcf9a4e1ab2f5cR49), so i'm just allowing all strings so all complex values get substituted.